### PR TITLE
Upgrade to new proxy to pick up Envoy cluster warming + Makefile fix

### PR DESCRIPTION
**Pull request by andraxylia**
Upgrade to new proxy to pick up Envoy cluster warming.

New Envoy also works around this failure:
[2018-03-15 18:13:33.252][10][warning][upstream] external/envoy/source/server/lds_subscription.cc:68] lds: fetch failure: gRPC client cluster 'mixer_check_server' is not static

Update dependency in Makefile, otherwise the build does not pick up the new Envoy (some recent changes were half-baked) and proxy upgrade did not work.

### CODE REVIEW

**Comment by ldemailly** [comment]
superseeds #4086 which does not pass tests either

File: istio.deps [hunk 1]
```diff
@@ -12,6 +12,6 @@
 		"repoName": "proxy",
 		"prodBranch": "master",
 		"file": "",
-		"lastStableSHA": "80ec4f6c2a672b00e55d34ef9b5d26a42a98da42"
+		"lastStableSHA": "9b4cbc6559822311d9bc04a6e0dd129b9b75617b"
```
Comment (JimmyCYJ) [inline istio.deps]: Could you update this to 9b4cbc6559822311d9bc04a6e0dd129b9b75617b? We would like to include that PR.

Comment (ldemailly) [reply istio.deps]: it already is that one :-)

Comment (JimmyCYJ) [reply istio.deps]: Thanks for correcting me.

**Comment by andraxylia** [comment]
yes, it's the only change that needs to be made to upgrade the proxy. There used to be another one in Docker.proxy, but that one got removed. Can I get a review approved?

**Comment by andraxylia** [comment]
circle-ci e2e pilot fails because of the known issue with routing rule v3.

**Comment by kyessenov** [comment]
Merge upstream and re-run circle ci tests.

**Comment by andraxylia** [comment]
@ZackButcher can you please approve? Need to review faster this kind of PRs.

**Comment by andraxylia** [comment]
merged with master, but still failing with v3, so that unrelated issue is not fixed.

**Comment by kyessenov** [comment]
v3 should not be run on master. Are you sure you re-run the test with the
latest?

**Comment by andraxylia** [comment]
I am sure I merged to master.

**Comment by kyessenov** [comment]
Looks like a legit error:
`I0315 21:39:46.038] 2018-03-15T21:39:46.037432Z	info	Start checking components' status
I0315 21:40:17.058] 2018-03-15T21:40:17.056726Z	info	Failed to setup framework: failed to get env ready: unable to comfirm compoment my_local_envoy is alive dial tcp [::1]:9090: connect: connection refused`

**Comment by costinm** [comment]
So it seems to be another 'prow specific' case - would really like to confirm this is a real bug captured only on GKE environment, or something specific to prow.

**Comment by ldemailly** [comment]
there has not been any "prow bug" outside of running out of quotas back in 0.2 time... so the first reaction should be, let's find out about the error, the burden should be to prove it's not a real issue/bug not the other way around

**Comment by andraxylia** [comment]
@kuat that error is from before the merge, not with the latest commit.

**Comment by andraxylia** [comment]
Why is prow not running?

**Comment by kyessenov** [comment]
@andraxylia it failed in the exact same spot -- it's likely a real failure.

**Comment by ldemailly** [comment]
```
I0316 00:26:09.354] 2018-03-16T00:26:09.354142Z	info	RunBackground: /home/bootstrap/go/src/istio.io/proxy/src/envoy/http/mixer/start_envoy -e /home/bootstrap/go/out/linux_amd64/release/envoy > /tmp/mixer_envoy_env743812681/my_local_proxy.log 2>&1
```
try
./tests/integration/example/integration.sh
(or
go test -v ./tests/integration/example/tests/sample1 -envoy_binary /home/bootstrap/go/out/linux_amd64/release/envoy -envoy_start_script /home/bootstrap/go/src/istio.io/proxy/src/envoy/http/mixer/start_envoy -mixer_binary /home/bootstrap/go/out/linux_amd64/release/mixs -fortio_binary fortio
)
to run it locally
(@yutongz will improve the script so logs show up in artifacts)

**Comment by andraxylia** [comment]
This tests is dangerous to run locally; it does a git reset --hard and some git clone ?

**Comment by andraxylia** [comment]
Not sure what the value of this test is, and what it is supposed to test. Is it that envoy can be run as a process? It copies envoy back and forth. Until this test is written to a standard where people can understand what the problem is, it adds little value and should be disabled @kyessenov

**Comment by andraxylia** [comment]
I run integration.sh manually and debugged it, and this template file from proxy used by the start_envoy script is outdated and does not work with the latest envoy:

https://github.com/istio/proxy/blob/master/src/envoy/http/mixer/envoy.conf.template

See attached output and generated envoy.conf file.

@qiwzhang can help bring that conf file up to date. Menawhile, the test should be disabled and this PR merged.

[envoy.conf.txt](https://github.com/istio/istio/files/1817932/envoy.conf.txt)
[start_envoy_output.txt](https://github.com/istio/istio/files/1817933/start_envoy_output.txt)

**Comment by ldemailly** [comment]
Thanks for figuring it out, that's great

> Meanwhile, the test should be disabled and this PR merged.

No I think given it works on master it should not be disabled but rather fixed
also we should maybe concerned about the incompatible change

**Comment by ldemailly** [comment]
ps: that template should be moved to istio/istio to avoid git cloning that repo just for that 1 file - that part maybe can wait

**Comment by andraxylia** [comment]
Meanwhile, we have to disable this test and remove the "git reset --hard" and all the other dangerous things, including the part where istio/proxy is re-downloaded. Hopefully people will not be tempted to run it locally and loose their changes. Tests should not alter the state of a workspace #4317


https://stackoverflow.com/questions/5788037/recover-from-git-reset-hard/5788069

**Comment by andraxylia** [comment]
@ldemailly this test is not ok, it should be disabled and removed until fixed properly.

**Comment by andraxylia** [comment]
That test should also do a make clean before make build, otherwise it uses old envoys from the /home/boot

**Comment by ldemailly** [comment]
please do file the issues you found with the test, but in the meantime we need the presubmit to pass without dropping the test (imo)

**Comment by ldemailly** [comment]
> This test is dangerous to run locally. It does a git reset --hard and some git clone.Luckily I did not have anything valuable in my ws.

there is always git reflog but the excerpt I mentioned above I took from the logs:
```
go test -v ./tests/integration/example/tests/sample1 -envoy_binary /home/bootstrap/go/out/linux_amd64/release/envoy -envoy_start_script /home/bootstrap/go/src/istio.io/proxy/src/envoy/http/mixer/start_envoy -mixer_binary /home/bootstrap/go/out/linux_amd64/release/mixs -fortio_binary fortio
```
isn't doing doing any of that, the `./tests/integration/example/integration.sh` I guess indeed appears to be geared more toward machine than human runs / not to be ran without looking

**Comment by andraxylia** [comment]
That test should not be a pre-submit, there is nothing in it to justify it as a pre-submit.

**Comment by sebastienvas** [comment]
@yutongz will work on removing the git reset part. I think we should wait until Wayne fixes the config to merge this. The point of this PR is to update Proxy, so if the current proxy is broken, it should not be merged.

**Comment by andraxylia** [comment]
@sebastienvas Wayne has fixed the config and I picked up the fix. I am waiting for the result. The failure of this test does not mean proxy is broken, it only means a file that only Wayne uses is out-of-date. So the test has little value and should not be in istio pre-submit, maybe it should be in istio/proxy presubmit.  I suggest we continue the discussion about the usefulness and correctness of that test in #4317 , there are too many things about that test.

**Comment by andraxylia** [comment]
@qiwzhang Did you merge without waiting for all the proxy tests? It appears it did not upload the envoy with that SHA, it is likely the tests that upload it to storage.

**Comment by sebastienvas** [comment]
@andraxylia SG

**Comment by andraxylia** [comment]
nvm, it seems there is a lag before Envoy is available

**Pull request merged**